hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -249,14 +251,20 @@ public void refreshTimeline() throws IOException {
     } else {
       this.commitTimelineOpt = Option.empty();
       String partitionColumns = HoodieWriterUtils.getPartitionColumns(keyGenerator);
-
+      LOG.warn("Partition cols " + partitionColumns);
       HoodieTableMetaClient.withPropertyBuilder()
           .setTableType(cfg.tableType)
           .setTableName(cfg.targetTableName)
           .setArchiveLogFolder(HOODIE_ARCHIVELOG_FOLDER_PROP.defaultValue())
           .setPayloadClassName(cfg.payloadClassName)
           .setBaseFileFormat(cfg.baseFileFormat)
           .setPartitionFields(partitionColumns)
+          .setPartitionFields(props.getProperty(DataSourceWriteOptions.PARTITIONPATH_FIELD().key()))
```

Comment:
Partition field is being set twice. Is this necessary?

Reply:
One of them is used for spark-sql and is a concatenation of partition columns. partitionField is used for virtual keys and is same value as "hoodie.datasource.write.partitionpath.field"